tests/sentry/api/endpoints/test_sentry_app_details.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import absolute_import
+
+from django.core.urlresolvers import reverse
+
+from sentry.testutils import APITestCase
+from sentry.mediators.sentry_apps import Creator
+from sentry.constants import SentryAppStatus
+
+
+class SentryAppDetailsTest(APITestCase):
+    def setUp(self):
+        self.superuser = self.create_user(email='a@example.com', is_superuser=True)
+        self.user = self.create_user(email='boop@example.com')
+        self.org = self.create_organization(owner=self.user)
+        self.super_org = self.create_organization(owner=self.superuser)
+        self.published_app = Creator.run(
+            name='Test',
+            user=self.user,
+            scopes=(),
+            webhook_url='https://example.com',
+        )
+        self.published_app.update(status=SentryAppStatus.PUBLISHED)
+        self.unpublished_app = Creator.run(
+            name='Testin',
+            user=self.user,
+            scopes=(),
+            webhook_url='https://example.com',
+        )
+        self.url = reverse('sentry-api-0-sentry-app-details', args=[self.published_app.id])
+
+    def test_superuser_sees_all_apps(self):
+        self.login_as(user=self.superuser)
+
+        with self.feature({
+            'organizations:internal-catchall': True,
+        }):
+            response = self.client.get(self.url, format='json')
+
+        assert response.status_code == 200
+        assert response.data == {
+            'name': self.published_app.name,
+            'scopes': [],
+            'uuid': self.published_app.uuid,
+            'webhook_url': self.published_app.webhook_url,
+        }
+
+        url = reverse('sentry-api-0-sentry-app-details', args=[self.unpublished_app.id])
+        with self.feature({
+            'organizations:internal-catchall': True,
+        }):
+            response = self.client.get(url, format='json')
+
+        assert response.status_code == 200
+        assert response.data == {
+            'name': self.unpublished_app.name,
+            'scopes': [],
+            'uuid': self.unpublished_app.uuid,
+            'webhook_url': self.unpublished_app.webhook_url,
+        }
+
+    def test_users_only_see_published_apps(self):
```

Comment:
How do normal users see their unpublished app?

Reply:
They don't, yet. "Unpublished" is effectively "development mode" where only those within the Org who owns it, can see it. The logic to enable the "only those within the Org" part of that will come in a followup PR. We're just trying to get the simple, happy-path, version of these endpoints in place first.